cognite/seismic/protos/v1/seismic_service_messages.proto

```diff
@@ -169,10 +169,19 @@ message VolumeRequest {
         Identifier seismic = 1;
         int64 seismic_store_id = 2;  // Need to be a data manager or tenant user, not 3rd party, to access by tracestore
     }
-    LineBasedVolume volume = 3;
+    oneof filter {
+        LineBasedVolume volume = 3;
+        GeometryBasedVolume geometry = 5;
+    }
```

Comment:
I didn't realize you could just put an existing field behind a oneof and be version-safe, but I guess that's how protobufs work!

Reply:
Protobufs are at heart just a concatenation of pairs of varint encoded tag numbers and values in a type-specific encoding. There's not even a defined ordering of fields in a message, this is explicitly left unspecified. This makes memory-efficient streaming serialization easier, but makes it impossible to reliably inspect the first N bytes of a large message to get a specific field, ie. for authenticating messages in a proxy or whatever. =)